src/main/java/org/opentripplanner/apis/gtfs/mapping/DirectModeMapper.java

```diff
@@ -0,0 +1,24 @@
+package org.opentripplanner.apis.gtfs.mapping;
+
+import org.opentripplanner.apis.gtfs.generated.GraphQLTypes;
+import org.opentripplanner.routing.api.request.StreetMode;
+
+/**
+ * Maps direct street mode from API to internal model.
+ */
+public class DirectModeMapper {
+
+  public static StreetMode map(GraphQLTypes.GraphQLPlanDirectMode mode) {
+    return switch (mode) {
+      case BICYCLE -> StreetMode.BIKE;
+      case BICYCLE_RENTAL -> StreetMode.BIKE_RENTAL;
+      case BICYCLE_PARKING -> StreetMode.BIKE_TO_PARK;
+      case CAR -> StreetMode.CAR;
+      case CAR_RENTAL -> StreetMode.CAR_RENTAL;
+      case CAR_PARKING -> StreetMode.CAR_TO_PARK;
```

Comment:
Does a direct mode of CAR_PARKING mean that you want to look for a parking lot near the destination and walk the rest?

Reply:
I think so yes. It's up to debate if we should support the functionality but pretty sure we do.